adapters/repos/db/shard.go

```diff
@@ -72,8 +72,8 @@ func NewShard(ctx context.Context, shardName string, index *Index) (*Shard, erro
 		deletedDocIDs:    docid.NewInMemDeletedTracker(),
 		cleanupInterval: time.Duration(index.invertedIndexConfig.
 			CleanupIntervalSeconds) * time.Second,
-		cleanupCancel: make(chan struct{}),
-		randomSource:  rand,
+		cancel:       make(chan struct{}, 1),
```

Comment:
why are you creating a buffered channel?

Reply:
If it is not buffered, the shutdown/drop is blocked indefinitely if there is no receiver. Right now the disk use scanner is the only receiver, so if the scan is not initiated (for example during a test, or if the user does not set the disk use config options), shutdown/drop cannot execute to completion.

Reply:
i actually had forgotten to add the logic to only scan if use explicitly passes the config to do so. just pushed up a moment ago. please let me know if you have any specific issues with creating a buffered channel here, or want to discuss more about it